dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -370,97 +373,253 @@ case class BinaryOutcomeDLCWithSelf(
+    setupDLC(messengerOpt).flatMap {
+      case SetupDLC(fundingTx,
+                    fundingSpendingInfo,
+                    cetWinLocal,
+                    cetLoseLocal,
+                    cetWinRemote,
+                    cetLoseRemote,
+                    _) =>
+        oracleSigF.flatMap { oracleSig =>
+          // Pick the CET to use and payout by checking which message was signed
+          val (cet, cetPrivKey, otherCetPrivKey) =
+            if (Schnorr.verify(messageWin, oracleSig, oraclePubKey)) {
+              if (local) {
+                (cetWinLocal, cetLocalWinPrivKey, cetRemoteWinPrivKey)
+              } else {
+                (cetWinRemote, cetRemoteWinPrivKey, cetLocalWinPrivKey)
+              }
+            } else if (Schnorr.verify(messageLose, oracleSig, oraclePubKey)) {
+              if (local) {
+                (cetLoseLocal, cetLocalLosePrivKey, cetRemoteLosePrivKey)
+              } else {
+                (cetLoseRemote, cetRemoteLosePrivKey, cetLocalLosePrivKey)
+              }
             } else {
-              finalRemotePrivKey
+              throw new IllegalStateException(
+                "Signature does not correspond to either possible outcome!")
             }
 
-            // Construct Closing Transaction
-            val txBuilder = BitcoinTxBuilder(
-              Vector(
-                TransactionOutput(output.value,
-                                  P2PKHScriptPubKey(finalPrivKey.publicKey))),
-              Vector(cetSpendingInfo),
-              feeRate,
-              changeSPK,
-              network
-            )
-
-            val spendingTxF = txBuilder.flatMap(subtractFeeAndSign)
-
-            spendingTxF.foreach(tx => logger.info(s"Closing Tx: ${tx.hex}"))
-
-            val spendingTxPublishedF = spendingTxF.flatMap { spendingTx =>
-              cetPublishedF.flatMap { _ =>
-                messengerOpt match {
-                  case Some(messenger) =>
-                    messenger
-                      .sendTransaction(spendingTx)
-                      .flatMap(_ => messenger.waitForConfirmations(blocks = 6))
-                  case None => FutureUtil.unit
+          val cetPublishedF = messengerOpt match {
+            case Some(messenger) =>
+              messenger
+                .sendTransaction(cet)
+                .flatMap(_ => messenger.waitForConfirmations(blocks = 6))
+            case None => FutureUtil.unit
+          }
+
+          // The prefix other refers to remote if local == true and local otherwise
+          val output = cet.outputs.head
+          val otherOutput = cet.outputs.last
+
+          // Spend the true case on the correct CET
+          val cetSpendingInfo = ConditionalSpendingInfo(
+            TransactionOutPoint(cet.txIdBE, UInt32.zero),
+            output.value,
+            output.scriptPubKey.asInstanceOf[ConditionalScriptPubKey],
+            Vector(cetPrivKey, ECPrivateKey(oracleSig.s)),
```

Comment:
this seems kind of hacky

Reply:
Yup, DLCs are essentially just this hack (assuming you are referring to `ECPrivateKey(oracleSig.s)`

Reply:
I would like to see if we can do something with the `Sign` interface, but i guess this isn't the time for that.